src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -324,21 +288,42 @@ private class TaskLauncherActor(
       promise.trySuccess(MatchedInstanceOps.noMatch(offer.getId))
 
     case ActorOfferMatcher.MatchOffer(offer, promise) =>
-      logger.debug(s"Matching offer ${offer.getId} and need to launch $instancesToLaunch tasks.")
-      val reachableInstances = instanceMap.filterNotAs{ case (_, instance) => instance.state.condition.isLost }
-      val matchRequest = InstanceOpFactory.Request(runSpec, offer, reachableInstances, instancesToLaunch, localRegion())
+      logger.info(s"Matching offer ${offer.getId} and need to launch $instancesToLaunch tasks.")
+      val reachableInstances = instanceMap.filterNotAs{
+        case (_, instance) => instance.state.condition.isLost || instance.isScheduled
+      }
+      val matchRequest = InstanceOpFactory.Request(runSpec, offer, reachableInstances, scheduledInstances, localRegion())
       instanceOpFactory.matchOfferRequest(matchRequest) match {
         case matched: OfferMatchResult.Match =>
-          logger.debug(s"Matched offer ${offer.getId} for run spec ${runSpec.id}, ${runSpec.version}.")
+          logger.info(s"Matched offer ${offer.getId} for run spec ${runSpec.id}, ${runSpec.version}.")
           offerMatchStatisticsActor ! matched
           handleInstanceOp(matched.instanceOp, offer, promise)
         case notMatched: OfferMatchResult.NoMatch =>
-          logger.debug(s"Did not match offer ${offer.getId} for run spec ${runSpec.id}, ${runSpec.version}.")
+          logger.info(s"Did not match offer ${offer.getId} for run spec ${runSpec.id}, ${runSpec.version}.")
           offerMatchStatisticsActor ! notMatched
           promise.trySuccess(MatchedInstanceOps.noMatch(offer.getId))
       }
   }
 
+  def syncInstances(): Unit = {
+    instanceMap = instanceTracker.instancesBySpecSync.instancesMap(runSpec.id).instanceMap
```

Comment:
I wonder if we could implement a simple subscribe mechanism and have the instanceTracker publish updates, so we wouldn't need the explicit sync

Reply:
My ideas is to make the `TaskLauncherActor` a pure function that retrieves a list of instances. See the prototype in [karsten/pass-instances](https://github.com/mesosphere/marathon/blob/karsten/pass-instances/src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala#L86). It does not remove the actor but shows how we can push the instances to the offer matching.